src/nvim/window.c

```diff
@@ -1928,9 +1928,11 @@ int win_close(win_T *win, bool free_buf)
         return FAIL;
     }
     win->w_closing = true;
-    apply_autocmds(EVENT_WINLEAVE, NULL, NULL, FALSE, curbuf);
-    if (!win_valid(win))
+    apply_autocmds(EVENT_WINLEAVE, NULL, NULL, false, curbuf);
```

Comment:
should be skipped if `recursive` (see other comments).

```c
if (!recursive) {
    apply_autocmds(EVENT_WINLEAVE, NULL, NULL, false, curbuf);
}
```

Reply:
Hi,

I do not see `recursive`  used anywhere in `window.c` nor in `buffer.c`, it is not even available from the current includes... where is it declared?

The code as it is now, tries to detect loops the way loops in other events are detected (checking if the window is still valid).